### PR TITLE
Blend new

### LINKED ISSUE #937: Fixing "blend" module and adding test
I tried using the Blend module and had trouble adding a custom function, so I wrote a test, but didn't quite finish it. Would anyone like to pick this up? I think the Blend module has a lot of promise for use in meta modules (now that those are working really well) and want to resuscitate it!

Any help appreciated!

### PR DESCRIPTION
**Pull request by keshav234156**
Fixes #937 #935 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/is-reviewers` for help, in a comment below
* [ ] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

File: src/modules/Blend/Module.js [hunk 1]
```diff
@@ -34,7 +31,8 @@ module.exports = function Dynamic(options, UI, util) {
 
     getPixels(priorStep.output.src, function(err, pixels) {
       options.firstImagePixels = pixels;
-
+      // convert to runnable code:
```
Comment (HarshKhandeparkar) [inline src/modules/Blend/Module.js]: Let's always add standardized comments from now ie Capital letter at the start and so on...

File: src/modules/Blend/Module.js [hunk 2]
```diff
@@ -43,7 +41,8 @@ module.exports = function Dynamic(options, UI, util) {
           p.get(x, y, 0),
           p.get(x, y, 1),
           p.get(x, y, 2),
-          p.get(x, y, 3)
+          p.get(x, y, 3),
+          x, y
```
Comment (HarshKhandeparkar) [inline src/modules/Blend/Module.js]: If all the other parameters are on newlines, let's keep the last two on newlines too. Let's make the code more systematic and easy to read.

**Comment by keshav234156** [comment]
@HarshKhandeparkar @jywarren In the blend module if we change the blend function and press apply button the output doesn't change, we have to reload to see the new output ,Don't know why it's happening? That's not the case with offset.

File: src/modules/Blend/info.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "blend": {
       "type": "string",
       "desc": "Function to use to blend the two images.",
-      "default": "function(r1, g1, b1, a1, r2, g2, b2, a2) { return [ r1, g2, b2, a2 ] }"
+      "default": "function(r1, g1, b1, a1, r2, g2, b2, a2,x,y) { return [ r2, g2, b2, a2 ] }"
```
Comment (HarshKhandeparkar) [inline src/modules/Blend/info.json]: Here too, please add spaces after the last two parameters because there are spaces in the first few.

**Comment by keshav234156** [comment]
@jywarren @HarshKhandeparkar Please review the changes!!!

**Comment by jywarren** [comment]
This looks awesome, and I'll go ahead and merge it! I'd love to protect this with a test as well if you're able to?

**Pull request merged**